core/.js/src/main/scala/io/udash/routing/UrlChangeProvider.scala

```diff
@@ -26,13 +26,13 @@ trait UrlChangeProvider {
   def onFragmentChange(callback: Url => Unit): Registration
 
   /** Changes the whole URL. */
-  def changeUrl(url: String): Unit
+  def changeUrl(url: String): Unit = dom.window.location.assign(url)
 }
 
 /** Used for routing based on the URL part following # sign. */
-object WindowUrlFragmentChangeProvider extends UrlChangeProvider {
+final class WindowUrlFragmentChangeProvider extends UrlChangeProvider {
```

Comment:
It could (and in some cases would be better), but this is better for dead-code elimination. Theoretically, we cannot remove the unused `object`, as its initialization might have a side effect - I'm not sure it always is removed, especially since IE/Edge initialize all objects eagerly.

Reply:
Got it.